crowbar_framework/spec/models/api/upgrade_spec.rb

```diff
@@ -89,7 +85,14 @@
         receive(:find).with("state:crowbar_upgrade AND NOT run_list_map:pacemaker-cluster-member").
         and_return([])
       )
-      expect(subject.services).to eq([:ok, ""])
+      allow_any_instance_of(NodeObject).to receive(:ssh_cmd).with(
+        "/usr/sbin/crowbar-shutdown-services-before-upgrade.sh"
+      ).and_return(true)
+      allow_any_instance_of(NodeObject).to receive(:ssh_cmd).with(
+        "/usr/sbin/crowbar-shutdown-services-before-upgrade.sh"
+      ).and_return(true)
```

Comment:
This is the kind of thing which ideally would belong in a separate PR - IIUC it's independent of the switch to class methods.